### PR TITLE
druid-shell: Added Delete key handling

### PR DESCRIPTION
**Pull request by bjorn**
As I was trying the examples on Linux I noticed that whereas Backspace was handled, the Delete key was not. I hope I found the right way to add this functionality (it works at least).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -684,6 +687,7 @@ Last release without a changelog :(
 [#1713]: https://github.com/linebender/druid/pull/1713
 [#1715]: https://github.com/linebender/druid/pull/1715
 [#1724]: https://github.com/linebender/druid/pull/1724
+[#1726]: https://github.com/linebender/druid/pull/1726
```
Comment (jneem) [inline CHANGELOG.md:690]: 1746, no?

**Review by jneem — CHANGES_REQUESTED** [review]
Thanks, this makes sense to me, at least. Just one typo in the changelog...

**Review by jneem — APPROVED** [review]

**Comment by cmyr** [comment]
Yep, this is definitely right; there's a bunch of stuff missing from this "simulate_input" method, and fixes are welcome!

**Comment by bjorn** [comment]
@cmyr I had just run into the missing Delete handling myself. What other stuff is known to be missing?

**Comment by cmyr** [comment]
I suspect most modifier combos like ctrl+arrow are missing. Also it might be nice, on linux, to support all the basic readline combos, like ctrl+e, ctrl+a, etcetera?

**Pull request merged**